src/Core/src/Layouts/GridLayoutManager.cs

```diff
@@ -1198,6 +1198,8 @@ class Definition
 		{
 			readonly GridLength _gridLength;
 			private double _size;
+			private readonly double _minSize;
```

Comment:
These should default to zero and Double.PositiveInfinity. The Definition constructor should also coerce each value to zero if something tries to set them less than zero. 

Then you avoid the extra ">= 0" check on each one during the Size setter.